proposals/devcontainer-id-variable.md

```diff
@@ -0,0 +1,60 @@
+## Goal
+
+Allow features to refer to an identifier that is unique to the dev container they are installed into and that is stable across rebuilds.
+
+E.g., the `docker-in-docker` feature needs a way to mount a volume per dev container.
+
+## Proposal
+
+The identifier will be referred to as `${devcontainerId}` in the feature metadata and that will be replaced with the dev container's id. It should only be used in parts of the metadata that is not used for building the image because that would otherwise prevent pre-building the image at a time when the dev container's id is not known yet. Excluding boolean and enum properties the properties supporting `${devcontainerId}` are: `entrypoint`, `mounts`, `customizations`.
+
+Implementations can choose how to compute this identifier. They must ensure that it is unique among other dev containers on the same Docker host and that it is stable across rebuilds of dev containers. The identifier must only contain alphanumeric characters. We describe a way to do this below.
+
+### Label-based Implementation 
+
+The following assumes that a dev container can be identified among other dev containers on the same Docker host by a set of labels on the container. Implementations may choose to follow this approach.
+
+The identifier is derived from the set of container labels uniquely identifying the dev container. It is up to the implementation to choose these labels, current experience suggests that the labels mainly refer to the dev container config. E.g., if the dev container is based on a local folder the label could be named `devcontainer.local_folder` and have the local folder's path as its value.
+
+E.g., the `docker-in-docker` feature could use the dev container id with:
```

Comment:
To avoid confusion with the syntax for feature ids (publishing syntax vs consuming syntax), it might be nice to point to the published docker-in-docker feature for this example instead of just referring to the name `docker-in-docker`.

```suggestion
E.g., the [`ghcr.io/devcontainers/features/docker-in-docker` feature](https://github.com/devcontainers/features/blob/main/src/docker-in-docker/devcontainer-feature.json) could use the dev container id with:
```